### DISCUSSION TITLE
Changing java package reference conventions for OracleTypes

### DISCUSSION THREAD

**Discussion by kalenp** [jeremyevans/sequel#2235] [General]
I'd like to change the references to OracleTypes [here](https://github.com/jeremyevans/sequel/blob/master/lib/sequel/adapters/jdbc/oracle.rb#L125) from the `Java::oracle.jdbc.OracleTypes::TIMESTAMPTZ` convention to `Java::OracleJdbc::OracleTypes::TIMESTAMPTZ`.

The reason for this is that our application loads the Oracle JDBC driver in a separate classloader, so when oracle.rb is loaded, these constants are not resolvable.  With the other JRuby convention for referencing packages, we are able to patch this in with an ad-hoc module that defines the types for Sequel to pick up.  It's not pretty, but it works.  With the current convention, though, you can't define `module oracle`, since `oracle` isn't a valid Ruby constant name.

There are several other places which use this package convention which could also be updated for consistency sake.  Overall, Sequel seems to use `Java::JavaSQL::` for the standard types, but the other convention for non-Java packages.

Would you be open to a PR to make these changes?

**Comment by jeremyevans** [comment]
Thank you for raising this as an potentially issue. I'd be open for a PR for these changes.  As long as it doesn't break existing tests, it should be acceptable.
